flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSourceFactory.java

```diff
@@ -124,110 +123,130 @@
 		params.putProperties(properties);
 
 		// validate
-		new SchemaValidator(true).validate(params);
+		// allow Kafka timestamps to be used, watermarks can not be received from source
+		new SchemaValidator(true, supportsKafkaTimestamps(), false).validate(params);
 		new KafkaValidator().validate(params);
-		formatValidator().validate(params);
 
-		// build
-		final KafkaTableSource.Builder builder = createBuilderWithFormat(params);
```

Comment:
We don't need to maintain a builder for an internal class in the future. The table sources should be immutable classes. If we add a new parameter then it should be added to the constructor so that all invocations need to be updated.